Gems/Atom/RHI/Code/Include/Atom/RHI/BufferPool.h

```diff
@@ -23,7 +23,59 @@ namespace AZ::RHI
         AZStd::unordered_map<int, void*> m_data;
     };
 
-    using BufferInitRequest = BufferInitRequestTemplate<Buffer>;
+    struct BufferInitRequest
```

Comment:
Lets add a comment to explain the purpose of this structure or how it will be used.